### PR TITLE
Add and apply scalafmt

### PR DESCRIPTION
**Pull request by mdedetrich**
Since there aren't any non trivial PR's open and I have wasted a non negligable amount of time manually formatting code to reduce diff's on PR's I think now is a good time to add scalafmt.

I have used a fairly conservative config, if you want me to update the config then just let me know and i can rebase the PR. Also when merging it do a "Rebase and merge" or "Merge commit" since I want to add the `Apply scalafmt` commit specifically to `.git-blame-ignore-revs` (which I will add in a future PR).

The PR also adds a github status check, you use this check to prevent PR's from being merged if they aren't formatted (just add the check to github branch settings).

### CODE REVIEW

File: .scalafmt.conf [hunk 1]
```diff
@@ -0,0 +1,11 @@
+version = 3.7.11
+runner.dialect = scala212
+preset = default
+align.preset = more
```
Comment (raboof) [inline .scalafmt.conf:4]: I don't really care for this alignment, can we stick to the default here?

File: build.sbt [hunk 2]
```diff
@@ -14,9 +14,8 @@ developers := List(
 
 scalaVersion := "2.12.18"
 
-/**
- * should work with later sbt versions as well (tested at least with 1.4.x)
- */
+/** should work with later sbt versions as well (tested at least with 1.4.x)
+  */
```
Comment (raboof) [inline build.sbt:18]: this should be either 3 lines or 1 line (maybe switch to `//`?), but not 2

File: src/main/scala/GpgHelpers.scala [hunk 3]
```diff
@@ -9,10 +9,7 @@ object GpgHelpers {
   val settings: Seq[Setting[_]] =
     Seq(
       packagedArtifacts := {
-      packagedArtifactsImpl(
-        packagedArtifacts.value,
-        gpg.value,
-        gpgWarnOnFailure.value)(streams.value.log.warn(_))
-    }
-  )
+        packagedArtifactsImpl(packagedArtifacts.value, gpg.value, gpgWarnOnFailure.value)(streams.value.log.warn(_))
```
Comment (raboof) [inline src/main/scala/GpgHelpers.scala:12]: this is not an improvement, but I guess we can live with that

File: .github/workflows/format.yml [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+on:
+  pull_request:
+    branches: ['**']
+
+jobs:
+  build:
+    name: Code is formatted
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout current branch (full)
```
Comment (raboof) [inline .github/workflows/format.yml:14]: WDYM `(full)`?

**Review by raboof — APPROVED** [review]
LGTM after switching the `align.preset` back to the default

File: .scalafmt.conf (hunk 1 above)
Comment (mdedetrich) [reply .scalafmt.conf]: I have updated the `.scalafmt.conf` and rebased/pushed PR with default alignment

File: .github/workflows/format.yml (hunk 4 above)
Comment (mdedetrich) [reply .github/workflows/format.yml:14]: I just copied this from Pekko, that text may be an artifact from an earlier revision which doesn't apply anymore

File: build.sbt (hunk 2 above)
Comment (mdedetrich) [reply build.sbt:18]: So there are various different doc types https://scalameta.org/scalafmt/docs/configuration.html#docstringsstyle, can you pick the style you like?

**Pull request merged**